device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -196,12 +188,6 @@ public DeviceClient setMessageCallback(MessageCallback callback, Object context)
     private DeviceClient(String uri, String deviceId, SecurityProvider securityProvider, IotHubClientProtocol protocol, ClientOptions clientOptions) throws URISyntaxException, IOException
     {
         super(uri, deviceId, securityProvider, protocol, SEND_PERIOD_MILLIS, getReceivePeriod(protocol), clientOptions);
-        commonConstructorSetup();
-    }
-
-    private void commonConstructorSetup()
-    {
-        this.ioTHubConnectionType = IoTHubConnectionType.SINGLE_CLIENT;
```

Comment:
Making this the default for the field at class instantiation time saves us from setting it in this method that was called by every constructor